File: OutOfOffice.Contracts/Models/LeaveRequest.cs
namespace OutOfOffice.Contracts.Models;

public class LeaveRequest
{
    public int Id { get; set; }
    public int EmployeeId { get; set; }
    public int AbsenceReasonId { get; set; }
    public DateTime StartDate { get; set; }
    public DateTime EndDate { get; set; }
    public int RequestTypeId { get; set; }
    public int? Hours { get; set; }
    public int StatusId { get; set; }
    public string Comment { get; set; }

    public AbsenceReason AbsenceReason { get; set; }
    public Employee Employee { get; set; }
    public RequestType RequestType { get; set; }
    public LeaveRequestStatus Status { get; set; }
}

File: outofoffice.client/src/app/models/leaveRequest.ts
import { Employee } from "./employee"

export interface LeaveRequest {
  id: number
  startDate: string
  endDate: string
  hours: number | null
  comment: string
  absenceReason: AbsenceReason
  employee: Employee
  requestType: RequestType
  status: LeaveRequestStatus
}

export interface AbsenceReason {
  id: number
  name: string
}

export interface RequestType {
  id: number
  name: string
}

export interface LeaveRequestStatus {
  id: number
  name: string
}

export interface LeaveRequestPayload {
  startDate: string
  endDate: string
  hours: number | null
  comment: string
  absenceReasonId: number
  employeeId: number
  requestTypeId: number
}
